### PR TITLE
ci: retry "vcpkg install google-cloud-cpp" once

### PR DESCRIPTION
**Pull request by devbww**
We have seen download errors here, so make a single retry after a
short pause. (See microsoft/vcpkg#11073 for a request to have
vcpkg do this itself.)

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5873)

### CODE REVIEW

File: ci/kokoro/docker/build-in-docker-quickstart-cmake.sh [hunk 1]
```diff
@@ -63,7 +63,12 @@ else
 fi
 
 "${vcpkg_bin}" remove --outdated --recurse
-"${vcpkg_bin}" install google-cloud-cpp
+if ! "${vcpkg_bin}" install google-cloud-cpp; then
+  # We have seen download errors here, so make a single retry after a
+  # short pause. See https://github.com/microsoft/vcpkg/issues/11073.
+  sleep 5
+  "${vcpkg_bin}" install google-cloud-cpp
```
Comment (devjgm) [inline ci/kokoro/docker/build-in-docker-quickstart-cmake.sh:70]: We have https://github.com/googleapis/google-cloud-cpp/blob/master/ci/retry-command.sh that we use in a few places. Could we use that here too?

Comment (devbww) [reply ci/kokoro/docker/build-in-docker-quickstart-cmake.sh]: Done.  I also tweaked `retry-command.sh` to act more like it advertises.  PTAL.

**Review by coryan — COMMENTED** [review]
> Reviewed 3 of 3 files at r2.
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5873)* status: all files reviewed, 2 unresolved discussions (waiting on @devbww and @devjgm)

___
*[ci/retry-command.sh, line 36 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5873#-MTrhI9j2EVe-kszt2om:-MTrhI9k5D2xZVHn88ba:by9u1dk) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/0824528d471f585433a7ad4ed4ed2b1730efc25a/ci/retry-command.sh#L36)):*
> ```Shell
> shift
> 
> "${PROGRAM}" "$@" && exit 0
> ```

Won't this fail immediately if `"${PROGRAM}" "$@"` fails? We do a `set -eu` at the top, which means the script aborts on the first failure?

**Comment by coryan** [comment]
___
*[ci/retry-command.sh, line 44 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5873#-MTrhgRcA_xf6pmIRujy:-MTrhgRcA_xf6pmIRujz:bsoc2tn) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/0824528d471f585433a7ad4ed4ed2b1730efc25a/ci/retry-command.sh#L44)):*
> ```Shell
>   sleep ${period}s
>   min_wait=$((min_wait * 2))
>   "${PROGRAM}" "$@" && exit 0
> ```

FYI, I do like the cleaner code!

**Review by devbww — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5873)* status: 0 of 3 files reviewed, 2 unresolved discussions (waiting on @coryan and @devjgm)

___
*[ci/retry-command.sh, line 36 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5873#-MTrhI9j2EVe-kszt2om:-MTs-rT1-3hW04KK9PYG:broiekj) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/0824528d471f585433a7ad4ed4ed2b1730efc25a/ci/retry-command.sh#L36)):*
<details><summary><i>Previously, coryan (Carlos O'Ryan) wrote…</i></summary><blockquote>

Won't this fail immediately if `"${PROGRAM}" "$@"` fails? We do a `set -eu` at the top, which means the script aborts on the first failure?
</blockquote></details>

No, I do not believe so.  Anything in a conditional context it not affected by `-e`.

**Review by devbww — COMMENTED** [review]
> Reviewed 3 of 3 files at r3.
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5873)* status: all files reviewed, 2 unresolved discussions (waiting on @coryan and @devjgm)

**Comment by coryan** [comment]
___
*[ci/retry-command.sh, line 36 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5873#-MTrhI9j2EVe-kszt2om:-MTs2rqdCwZRGKfBZgnm:b4m8pkv) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/0824528d471f585433a7ad4ed4ed2b1730efc25a/ci/retry-command.sh#L36)):*
<details><summary><i>Previously, devbww (Bradley White) wrote…</i></summary><blockquote>

No, I do not believe so.  Anything in a conditional context it not affected by `-e`.
</blockquote></details>

TIL, thanks.

**Review by devbww — COMMENTED** [review]
> Dismissed @coryan from a discussion.
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5873)* status: all files reviewed, 1 unresolved discussion (waiting on @devjgm)

**Review by coryan — APPROVED** [review]
:lgtm: I did not realize this was blocked.

> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5873#-:-MTsKeIT4Nf0xH8CmGFp:b-rz3ar7)* status: all files reviewed, 1 unresolved discussion (waiting on @devjgm)

**Review by devbww — COMMENTED** [review]
I don't know what I did to elicit that "dismiss" thing, but I was going to wait for @devjgm anyway.  Thanks.

> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5873#-:-MTsNelH1rlnVZqWnvnS:b-1y966s)* status: all files reviewed, 1 unresolved discussion (waiting on @devjgm)

**Review by devjgm — APPROVED** [review]

**Pull request merged**